test/e2e/monitoring/alerts_test.go

```diff
@@ -139,5 +147,82 @@ var _ = Context("Prometheus Alerts", func() {
 				ScaleDeployment(components.Name, components.Namespace, 1)
 			})
 		})
+
+		Context("and there are duplicate MACs", func() {
+			var virtClient kubecli.KubevirtClient
+			var err error
+
+			BeforeEach(func() {
+				By("starting virtClient")
+				virtClient, err = kubecli.GetKubevirtClientFromFlags("", os.Getenv("KUBECONFIG"))
+				Expect(err).ToNot(HaveOccurred())
+
+				By("creating test namespace")
+				namespace := &k8sv1.Namespace{ObjectMeta: metav1.ObjectMeta{
+					Name: kvtutil.NamespaceTestDefault,
+					Labels: map[string]string{
+						"mutatevirtualmachines.kubemacpool.io": "ignore",
+					},
+				}}
+				_, err = virtClient.CoreV1().Namespaces().Create(context.Background(), namespace, metav1.CreateOptions{})
+				Expect(err).ToNot(HaveOccurred())
+			})
+
+			AfterEach(func() {
+				By("deleting test namespace")
+				err = virtClient.CoreV1().Namespaces().Delete(context.Background(), kvtutil.NamespaceTestDefault, metav1.DeleteOptions{})
+				Expect(err).ToNot(HaveOccurred())
+			})
+
+			It("should issue KubeMacPoolDuplicateMacsFound alert", func() {
+				By("creating 2 VMs with a duplicate MAC")
+				err = createVirtualMachine(virtClient)
+				Expect(err).ToNot(HaveOccurred())
+				err = createVirtualMachine(virtClient)
+				Expect(err).ToNot(HaveOccurred())
+
+				By("cleaning namespace labels")
+				err = cleanNamespaceLabels(virtClient, kvtutil.NamespaceTestDefault)
+				Expect(err).ToNot(HaveOccurred())
+
+				By("restaring kubemacpool pods")
+				restartKubemacpoolPods(virtClient)
+
+				By("waiting for the amount of time it takes the alert to fire")
+				time.Sleep(5 * time.Minute)
+				By("checking existence of alert")
+				prometheusClient.checkForAlert("KubeMacPoolDuplicateMacsFound")
+			})
+		})
 	})
 })
+
+func createVirtualMachine(virtClient kubecli.KubevirtClient) error {
```

Comment:
please add the MAC Address as a input param, so that in the test it would be apparent that we are indeed creating 2 VMs with the same MAC Address.
Also please change the function name to `createVirtualMachineWithPrimaryInterfaceMacAddress` or something..

Reply:
changed